src/poetry/puzzle/provider.py

```diff
@@ -725,6 +726,7 @@ def debug(self, message: str, depth: int = 0) -> None:
         if message.startswith("fact:"):
             if "depends on" in message:
                 m = re.match(r"fact: (.+?) depends on (.+?) \((.+?)\)", message)
+                assert m is not None
```

Comment:
this may not be preferable

Reply:
I think we'd rather throw an exception here -- while it's a good insight from mypy that there is no error handling here and we implicitly make this assertion, I would like to avoid use of the `assert` keyword.

Reply:
makes sense, chose to `raise ValueError`. lmk if you want to see something different